highlevel/robot/gateway/motor.py

```diff
@@ -1,73 +1,156 @@
+        Applies a scaling factor to each value.
+        """
+        scale_factor = self.configuration.pid_scale_factor
+        update_rate = self.configuration.motor_update_rate
+        max_value_on_motor_board = (2**32) - 1
+        if pid_constants.k_i * scale_factor / update_rate >= max_value_on_motor_board:
+            LOGGER.get().error('pid_to_proto_ki_too_large',
+                               k_i=pid_constants.k_i)
+        if pid_constants.k_d * scale_factor * update_rate >= max_value_on_motor_board:
+            LOGGER.get().error('pid_to_proto_kd_too_large',
+                               k_d=pid_constants.k_d)
+
+        return PIDCoefficients(
+            kp=pid_constants.k_p,
+            ki=pid_constants.k_i,
+            kd=pid_constants.k_d,
+        )
+
+    async def _send_pid_configs(self) -> None:
+        """
+        Sends last updated PID configurations.
+        """
+        message = BusMessage(pidConfig=PIDConfigMsg(
+            pid_speed_left=self._pid_constants_to_proto(self.pid_speed_left),
+            pid_speed_right=self._pid_constants_to_proto(self.pid_speed_right),
+            pid_position_left=self._pid_constants_to_proto(
+                self.pid_position_left),
+            pid_position_right=self._pid_constants_to_proto(
+                self.pid_position_right),
+        ))
+        await self._send_message(message)
+
+    async def set_pwms(self, ratio_left: float, ratio_right: float) -> None:
+        """
+        Sets each wheel's PWM output. Ratios are duty cycles, signs are directions.
+        """
+        LOGGER.get().info('motor_gateway_set_pwms',
+                          ratio_left=ratio_left,
+                          ratio_right=ratio_right)
+        message = BusMessage(wheelPWM=WheelPWMMsg(ratio_left=ratio_left,
+                                                  ratio_right=ratio_right))
+        await self._send_message(message)
+
+    async def set_target_speeds(self, tick_left: TickPerSec,
+                                tick_right: TickPerSec) -> None:
         """
         Sets each wheel's speed target.
         """
-        LOGGER.get().debug('gateway_set_speed',
-                           left_speed=left_speed,
-                           right_speed=right_speed)
+        LOGGER.get().debug('motor_gateway_set_target_speeds',
+                           left_speed=tick_left,
+                           right_speed=tick_right)
         message = BusMessage(moveWheelAtSpeed=MoveWheelAtSpeedMsg(
-            left_tick_per_sec=round(left_speed),
-            right_tick_per_sec=round(right_speed)))
+            left_tick_per_sec=round(tick_left),
+            right_tick_per_sec=round(tick_right)))
         await self._send_message(message)
 
-    async def send_pid(self, pid_left: PIDValues,
-                       pid_right: PIDValues) -> None:
+    async def set_target_positions(self, tick_left: Tick,
```

Comment:
do we still need that function?

Reply:
We will, when doing an action the robot needs to stay still.

Reply:
what's the difference bw using this function and set_speed(0,0) when we need the robot to stay still?